src/main/java/de/tum/in/www1/artemis/web/rest/AchievementResource.java

```diff
@@ -0,0 +1,135 @@
+
+    /**
+     * GET /achievements : Get all achievements for user.
+     *
+     * @return the ResponseEntity with status 200 (OK) and with the found achievements as body
+     */
+    @GetMapping("/achievements")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Set<Achievement>> getAchievementsForUser() {
+        User user = userService.getUser();
+        log.debug("REST request to get achievements for user : {}", user.getLogin());
+        Set<Achievement> achievements = achievementService.findAllForUser(user.getId());
+        return ResponseEntity.ok(achievements);
+    }
+
+    /**
+     * GET /courses/{courseId}/achievements : Get all achievements for course.
+     *
+     * @param courseId the course to which the achievements belong to
+     * @return the ResponseEntity with status 200 (OK) and with the found achievements as body
+     */
+    @GetMapping("/courses/{courseId}/achievements")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Set<Achievement>> getAchievementsForCourse(@PathVariable Long courseId) {
+        log.debug("REST request to get achievements for course : {}", courseId);
+        Set<Achievement> achievements = achievementService.findAllForCourse(courseId);
+        return ResponseEntity.ok(achievements);
+    }
+
+    /**
+     * PUT /achievements : Updates an existing achievement.
+     *
+     * @param achievement the achievement to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated achievement, or with status 500
+     *         (Internal Server Error) if the achievement couldn't be updated
+     */
+    @PutMapping("/achievements")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Achievement> updateAchievement(@RequestBody Achievement achievement) {
+        log.debug("REST request to update Achievement : {}", achievement);
+
+        if (achievement.getId() == null) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        Course course = achievement.getCourse();
+        User user = userService.getUserWithGroupsAndAuthorities();
+        if (!authCheckService.isAtLeastInstructorInCourse(course, user)) {
+            throw new AccessForbiddenException("You are not allowed to access this resource");
+        }
+
+        Achievement savedAchievement = achievementRepository.save(achievement);
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, savedAchievement.getId().toString())).body(savedAchievement);
+    }
+
+    /**
+     * DELETE /achievements/:achievementId : delete the "id" achievement.
+     *
+     * @param achievementId the id of the achievement to delete
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @DeleteMapping("/achievements/{achievementId}")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    @Transactional
```

Comment:
please remove Transactional, we don't want to use it!

Reply:
removed